### PR TITLE
Add a new user script for calculating ellipsoidal area for a polygon layer

### PR DESCRIPTION
**Pull request by spatialthoughts**
Calculates the area of each feature using the given ellipsoid. The area is added as a new attribute to the output layer.

### CODE REVIEW

**Comment by spatialthoughts** [comment]
@anitagraser Can you review and merge this?

**Comment by anitagraser** [comment]
@spatialthoughts sorry, I don't have commit rights here either.

ping @slarosa @volaya

File: scripts/ellipsoidal_area.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+##input=vector polygon
```
Comment (slarosa) [inline scripts/ellipsoidal_area.py]: could you add a name to the top (i.e ```##name=Ellipsoid Area```) and a group (maybe Utils)?

File: scripts/ellipsoidal_area.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+##ellipsoid=string WGS84
+##new_field=string Area
+##units=string sq_km
```
Comment (slarosa) [inline scripts/ellipsoidal_area.py]: maybe better to use ```selection``` token here, so the user can choose units by a dropdown menu.

File: scripts/ellipsoidal_area.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  elif units == 'sq_nm':
+    final_area = area.convertMeasurement(
+        polygon_area, QGis.Meters, QGis.NauticalMiles, True)[0]
+  elif units == 'sq_degrees':
+    final_area = area.convertMeasurement(
+        polygon_area, QGis.Meters, QGis.Degrees, True)[0]
+  else:
+    final_area = polygon_area
+
+  attrs = feat.attributes() + [final_area]
```
Comment (slarosa) [inline scripts/ellipsoidal_area.py]: you could use ```attrs.append(final_area)```

**Comment by slarosa** [comment]
hi @spatialthoughts I added some comments to the code and I tested the script and works fine, thanks

**Comment by spatialthoughts** [comment]
Thanks @slarosa for the review and feedback. I have addressed your comments and pushed a new version. Please have a look.

**Comment by slarosa** [comment]
Thanks @spatialthoughts and sorry for the delay, this PR was  merged in master (60df2ae and b565c4b)

**Pull request closed without merging**